README.md

```diff
@@ -0,0 +1,900 @@
+
+> [!WARNING]
+> By definition `"/"` implies all of the commands available on a resource, and SHOULD be used with great care.
+
+# Policy
+[Policy]: #policy
+
+UCAN Delegation uses predicate logic statements extended with [jq]-inspired selectors as a policy language. Policies are syntactically driven, and MUST constrain the `args` field of an eventual [Invocation].
+
+A Policy is always given as an array of predicates. This top-level array is implicitly treated as a logical `and`, where `args` MUST pass validation of every top-level predicate.
+
+Policies are structured as trees. With the exception of subtrees under `any`, `or`, and `not`, every leaf MUST evaluate to `true`. 
+
+A Policy is an array of statements. Every statement MUST take the form `[operator, selector, argument]` except for negation which MUST take the form `["not", statement]`.
+
+Below is a formal syntax for the UCAN Policy Language given in [ABNF] (representing IPLD as [DAG-JSON]):
+
+``` abnf
+policy      = "[" *1(statement *("," statement)) "]"
+statement   = equality 
+            / inequality 
+            / wildcard
+            / connective 
+            / quantifier
+
+;; STRUCTURAL
+
+connective  = "[" DQUOTE "not" DQUOTE ","  statement "]"                   ; Negation
+            / "[" DQUOTE "and" DQUOTE ",[" statement *("," statement) "]]" ; Conjuction
+            / "[" DQUOTE "or"  DQUOTE ",[" statement *("," statement) "]]" ; Disjunction
+
+quantifier  = "[" DQUOTE "all" DQUOTE "," selector "," policy "]" ; Universal
+            / "[" DQUOTE "any" DQUOTE "," selector "," policy "]" ; Existential
+
+;; COMPARISONS
+
+equality    = "[" DQUOTE "==" DQUOTE "," selector "," ipld   "]" ; Equality on IPLD literals
+inequality  = "[" DQUOTE ">"  DQUOTE "," selector "," number "]" ; Numeric greater-than
+            / "[" DQUOTE ">=" DQUOTE "," selector "," number "]" ; Numeric greater-than-or-equal
+            / "[" DQUOTE "<"  DQUOTE "," selector "," number "]" ; Numeric lesser-than
+            / "[" DQUOTE "<=" DQUOTE "," selector "," number "]" ; Numeric lesser-than-or-equal
+
+wildcard    = "[" DQUOTE "like" DQUOTE "," selector "," pattern "]" ; String wildcard matching
+
+;; SELECTORS
+
+selector    = DQUOTE "." DQUOTE                     ; Identity
+            / DQUOTE 1*(subselector *1("?")) DQUOTE ; Nested subselectors with possible optionals
+
+subselector = "." CHAR string                           ; Dotted field selector
+            / *1(".") "[\" DQUOTE string "\" DQUOTE "]" ; Explicit field selector
+            / *1(".") "[" integer "]"                   ; Index selector
+            / *1(".") "[]"                              ; Collection values // FIXME doble check code
+
+;; SPECIAL LITERALS
+
+pattern     = DQUOTE string DQUOTE ; Reminder: IPLD strings are UTF-8
+number      = integer / float
```

Comment:
According to this line, selectors only operates on integers or floats, but below the spec also mention selecting on bytes (`Selecting on Bytes` section). Should bytes be included here?

Reply:
nevermind, I got it wrong